### PR TITLE
chore: adding scripts and workflow

### PR DESCRIPTION
**Pull request by kaltepeter**
Run and test will not work until we have a test folder and test image.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,6 +1,8 @@
 FROM --platform=$BUILDPLATFORM python:3.11-slim-bookworm
 ARG TARGETPLATFORM
 ARG BUILDPLATFORM
+ARG TARGETARCH
+RUN echo "I am running on $BUILDPLATFORM, building for $TARGETPLATFORM with $TARGETARCH"
```
Comment (kaltepeter) [inline Dockerfile:5]: this is helpful for debugging.

File: Dockerfile [hunk 2]
```diff
@@ -26,5 +28,6 @@ pipx     $(pipx --version)
 poetry   $(poetry --version)
 EOF
 
+EXPOSE 8000
```
Comment (kaltepeter) [inline Dockerfile:31]: default fastapi port for convience.

File: Dockerfile [hunk 3]
```diff
@@ -1,15 +1,43 @@
...
 EOF
 
+RUN <<EOF
+    set -eux
+    apt-get update
+    apt-get install --assume-yes --no-install-recommends pipx \
+    dumb-init \
+    nodejs \
+    npm \
+    unzip \
```
Comment (kaltepeter) [inline Dockerfile:20]: adding debug/ci tools

File: Dockerfile [hunk 4]
```diff
@@ -1,15 +1,43 @@
...
+    curl \
+    git \
+    wget
+
+    rm -rf /var/lib/apt/lists/*
+    apt-get clean
+EOF
+
+# TODO: fix
+RUN <<EOF
```
Comment (NickBeeuwsaert-Datasite) [inline Dockerfile:32]: do we need this to be separate from the previous RUN?

Comment (kaltepeter) [reply Dockerfile:32]: we could potentially combine these, but it's so different of a process from simple apt-get install and the layer cache frequency is different. It's also not working yet. Maybe we can combine when I fix it?

Comment (kaltepeter) [reply Dockerfile]: scratch that. It was working, I needed to force a pull with my testing. I combined them.

**Review by NickBeeuwsaert-Datasite — APPROVED** [review]

**Pull request merged**